pkg/maps/ipcache/ipcache.go

```diff
@@ -262,24 +265,34 @@ func (m *Map) supportsDelete() bool {
 // SupportsDelete determines whether the underlying kernel map type supports
 // the delete operation.
 func SupportsDelete() bool {
-	return IPCache.supportsDelete()
+	return GetIPCache().supportsDelete()
 }
 
 // BackedByLPM returns true if the IPCache is backed by a proper LPM
 // implementation (provided by Linux kernels 4.11 or later), false otherwise.
 func BackedByLPM() bool {
-	return IPCache.MapType == bpf.MapTypeLPMTrie
+	return GetIPCache().MapType == bpf.MapTypeLPMTrie
 }
 
 var (
 	// IPCache is a mapping of all endpoint IPs in the cluster which this
 	// Cilium agent is a part of to their corresponding security identities.
 	// It is a singleton; there is only one such map per agent.
-	IPCache = NewMap(Name)
+	ipcache *Map
+	once    = &sync.Once{}
 )
 
+// GetIPCache gets the ipcache Map singleton. If it has not already been done,
```

Comment:
nit: `IPCacheMap()` perhaps, to keep naming consistent with the other maps?

Reply:
good idea